playbooks/roles/ocp-cost-management/tasks/cmo-test.yml

```diff
@@ -0,0 +1,112 @@
+---
+- name: Setup and Test Cost Management Operator in OpenShift
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: Log in to OpenShift
+      ansible.builtin.shell: >
+        oc login -u kubeadmin -p {{ openshift_console_password }}
+      environment:
+        KUBECONFIG: "/root/openstack-upi/auth/kubeconfig"
+      become: yes
+
+    - name: Patch OperatorHub to disable default sources
+      ansible.builtin.command:
+        cmd: >
+          oc patch operatorhub.config.openshift.io/cluster
+          -p='{"spec":{"disableAllDefaultSources":true}}'
+          --type=merge
+      become: yes
+
+    - name: Apply ImageContentSourcePolicy
+      ansible.builtin.k8s:
+        state: present
+        definition:
+          apiVersion: operator.openshift.io/v1alpha1
+          kind: ImageContentSourcePolicy
+          metadata:
+            name: brew-registry
+          spec:
+            repositoryDigestMirrors:
+              - mirrors:
+                  - brew.registry.redhat.io
+                source: registry.redhat.io
+              - mirrors:
+                  - brew.registry.redhat.io
+                source: registry.stage.redhat.io
+              - mirrors:
+                  - brew.registry.redhat.io
+                source: registry-proxy.engineering.redhat.com
+
+    - name: Create custom CatalogSource for Cost Management
+      ansible.builtin.k8s:
+        state: present
+        definition:
+          apiVersion: operators.coreos.com/v1alpha1
+          kind: CatalogSource
+          metadata:
+            name: my-operator-catalog
+            namespace: openshift-marketplace
+          spec:
+            sourceType: grpc
+            publisher: grpc
+            displayName: my-operator-catalog
+            image: brew.registry.redhat.io/rh-osbs/iib:{{ cost_management_catalogsource_image}}

```

Comment:
to create ImageContentSourcePolicy and custom catalogsource use the "set-custom-catalogsource" role